packages/react-utilities/src/hooks/useTriggerElement.ts

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+
+import { applyTriggerPropsToChildren } from '../utils/applyTriggerPropsToChildren';
+import { getReactCallbackName } from '../utils/getReactCallbackName';
+import { onlyChild } from '../utils/onlyChild';
+import { useEventCallback } from './useEventCallback';
+import { useMergedRefs } from './useMergedRefs';
+import type { ReactCallbackName } from '../utils/getReactCallbackName';
+
+export type UseTriggerElementOptions<TriggerProps> = {
+  /** An actual trigger element or render props function. */
+  children: React.ReactElement | ((props: TriggerProps) => React.ReactNode) | null | undefined;
+
+  /** A ref to trigger element. */
+  ref: React.Ref<unknown> | undefined;
+
+  /** Props that are passed to a parent component and should be forwarded down. */
+  outerProps: React.HTMLProps<unknown>;
```

Comment:
nit: it is not clear what `outerProps` mean without reading the docs, `passedProps` would be more understandable for me, but to be fair, it seems other libraries use **outer** as well.

Reply:
https://github.com/microsoft/fluentui/pull/21225#discussion_r781338162
Will be happy to iterate on interface there.